From 2a8298a27e1313af552c049e35ec8c8c82197285 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Mon, 17 Dec 2018 22:00:24 +0100
Subject: grub-core/loader/efi/fdt.c: do not copy random memory

We should not try to copy any memory area which is outside of the original
fdt. If this extra memory is controlled by a hypervisor this might end
with a crash.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>

Origin: other, https://lists.gnu.org/archive/html/grub-devel/2018-12/msg00042.html
Last-Update: 2018-12-21

Patch-Name: efi-fdt-do-not-copy-random-memory.patch
---
 grub-core/loader/efi/fdt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/grub-core/loader/efi/fdt.c b/grub-core/loader/efi/fdt.c
index a4c6e8036..c9aee74ef 100644
--- a/grub-core/loader/efi/fdt.c
+++ b/grub-core/loader/efi/fdt.c
@@ -66,7 +66,7 @@ grub_fdt_load (grub_size_t additional_size)
 
   if (raw_fdt)
     {
-      grub_memmove (fdt, raw_fdt, size);
+      grub_memmove (fdt, raw_fdt, size - additional_size);
       grub_fdt_set_totalsize (fdt, size);
     }
   else
